pkg/model/loader.go

```diff
@@ -86,21 +111,29 @@ func (ml *ModelLoader) TemplatePrefix(modelName string, in interface{}) (string,
 	return buf.String(), nil
 }
 
-func (ml *ModelLoader) loadTemplateIfExists(modelName, modelFile string) error {
+func (ml *ModelLoader) TemplatePrefix(templateName string, in PromptTemplateData) (string, error) {
+	return evaluateTemplate[PromptTemplateData](templateName, in, ml.ModelPath, &(ml.promptsTemplates), &(ml.mu))
+}
+
+func (ml *ModelLoader) TemplateForChatMessage(templateName string, messageData ChatMessageTemplateData) (string, error) {
+	return evaluateTemplate[ChatMessageTemplateData](templateName, messageData, ml.ModelPath, &(ml.chatMessageTemplates), &(ml.mu))
```

Comment:
just wondering, why passing the pointer to the map around?

Reply:
Passing the pointer to the map is because evaluateTemplate[T] was generic - See other comment for style choice on why this line in general is generic. 

If that function is generic, it can't accept an `*ModelLoader` receiver, so I just have it accept a pointer to the relevant lock and map... admittedly, a hack, but it's "contained" to the private method here in this file at least.

If we de-generic that method, I would presumably remove it.